### PR TITLE
[MPA] Class-Incremental Learning

### PR DESCRIPTION
**Pull request by goodsong81**
### What's in this PR
* New algo backend: external/model-preparation-algorithm
  * submodule -> https://github.com/openvinotoolkit/model_preparation_algorithm
* Task implementation & model templates
  * Class-Incremental learning for Classification
  * Class-Incremental learning for Detection
  * Class-Incremental learning for Segmentation
* Sample codes (for each task)
* CLI tests (for each task)
* Change proposal for common CLI & test function
  * Label schema loading for pretrained model
  * Optional --load-weights argument for `otd_train_testing()`
### How to test
* Env setup
```console
$ cd $OTE_ROOT/external/model-preparation-algorithm
$ CUDA_HOME=/usr/local/cuda-11.1 CUDA_VERSION=11.1 ./init_venv.sh ./venv
$ . ./venv/bin/activate
```
* Run sample
```console
$ cd $OTE_ROOT/external/model-preparation-algorithm
$ python mpa_tasks/samples/det_cls_il_sample.py configs/detection/mobilenetv2_atss_cls_incr/template.yaml
```
* Run tests
```console
$ cd $OTE_ROOT
$ CUDA_VISIBLE_DEVICES=0 pytest -rP tests/ote_cli/external/model-preparation-algorithm/ote_cli
```

### CODE REVIEW

File: ote_cli/ote_cli/tools/train.py [hunk 1]
```diff
@@ -138,12 +145,20 @@ def main():
...
+                    )
+                }
+            )
         environment.model = ModelEntity(
             train_dataset=dataset,
             configuration=environment.get_model_configuration(),
-            model_adapters={
-                "weights.pth": ModelAdapter(read_binary(args.load_weights))
-            },
+            model_adapters=model_adapters,
```
Comment (goodsong81) [inline ote_cli/ote_cli/tools/train.py]: It is natural to load label schema on which the model weights had been trained, if any. Class-Incremental learning exploits the gap between the global schema and the model schema.

File: ote_cli/ote_cli/utils/tests.py [hunk 2]
```diff
@@ -270,7 +272,7 @@ def ote_deploy_openvino_testing(template, root, ote_dir, args):
         deployment_dir,
     ]
     assert run(command_line, env=collect_env_vars(work_dir)).returncode == 0
-    assert run(["unzip", "openvino.zip"], cwd=deployment_dir).returncode == 0
+    assert run(["unzip", "-o", "openvino.zip"], cwd=deployment_dir).returncode == 0
```
Comment (goodsong81) [inline ote_cli/ote_cli/utils/tests.py:281]: 'Overwrite' option helped resolve failure due to existing unzipped files in case of multiple run.

File: ote_cli/ote_cli/utils/tests.py [hunk 3]
```diff
@@ -100,6 +100,8 @@ def ote_train_testing(template, root, ote_dir, args):
         "--save-model-to",
         f"{template_work_dir}/trained_{template.model_template_id}",
     ]
+    if "--load-weights" in args:
+        command_line.extend(["--load-weights", f'{os.path.join(ote_dir, args["--load-weights"])}'])
     command_line.extend(args["train_params"])
```
Comment (goodsong81) [inline ote_cli/ote_cli/utils/tests.py]: I think tests for model training need to adopt --load-weights argument if specified.

**Comment by goodsong81** [comment]
run ote-test

**Comment by goodsong81** [comment]
run ote_sdk tests

**Comment by goodsong81** [comment]
run ote_sdk tests

**Comment by goodsong81** [comment]
run ote-test

**Comment by goodsong81** [comment]
run ote-test

**Comment by goodsong81** [comment]
run ote-test

**Comment by goodsong81** [comment]
run ote-test

**Comment by goodsong81** [comment]
run ote-test

**Comment by goodsong81** [comment]
Tested: https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-ote-test/142/
Failures are not relevant to this PR.
